File: client/src/types/analytics.ts
export interface TopicPerformance {
  topic: string;
  score: number;
}

export interface WeeklyProgress {
  week: string;
  score: number;
}

export interface Analytics {
  totalPDFs: number;
  totalChats: number;
  totalTests: number;
  averageScore: number;
  studyHours: number;

  topicPerformance: TopicPerformance[];

  weeklyProgress: WeeklyProgress[];
}

File: server/app/schemas/analytics.py
from pydantic import BaseModel


class AnalyticsResponse(BaseModel):
    totalPDFs: int
    totalChats: int
    totalTests: int
    averageScore: float
    studyHours: float
    topicPerformance: list[dict]
    weeklyProgress: list[dict]


class RecommendationItem(BaseModel):
    id: str
    title: str
    description: str
    priority: str
    topic: str


class StudyPlanTask(BaseModel):
    id: str
    topic: str
    duration: str
    completed: bool


class RecommendationResponse(BaseModel):
    recommendations: list[RecommendationItem]
    studyPlan: list[StudyPlanTask]
